src/components/ec-table-filter/ec-table-filter.story.js

```diff
@@ -280,6 +308,20 @@ const AllTemplate = storyArgs => ({
         />
       </div>
     </div>
+    <h2 class="tw-m-24">Use custom CSS Grid layout</h2>
+    <div class="tw-flex tw-px-20">
+      <div class="tw-my-auto tw-mx-20 tw-w-full ec-card">
+        <ec-table-filter
+          v-model="model"
+          :filters="searchFilterFillingRemainingSpace"
+          layout-class="!tw-grid tw-grid-rows-2 !tw-grid-flow-row tw-grid-cols-[repeat(4,_auto)_1fr_auto] lg:tw-grid-rows-1"
```

Comment:
This violates basic principles that parent should not dictate how a child should be displayed. I'm still lost why is this needed?

Reply:
The idea is to let the user of the `ec-table-filter` component to style the way filters are being displayed. In this case is used to move the search filter bar below the other filters when needed, but if we want to use the same component but with a different layout the idea was using this prop instead of modifying the component at all

Reply:
We should pass a boolean `stretch` to the filter and ec-table-filter should determine the layout based on that. If it detects that all filters in the `filters` prop have stretch set to false, use one layout, otherwise use different layout.

Reply:
In this case I suggested this approach to let parent component to set grid layout. Yes it's not the ideal solution, but to have flexibility depends on a particular use case we can use it.

We share a restricted part of knowledge to the parent component. Just grid layout information.

I wanted to avoid situation when tomorrow we'll be asked to show all filters one under another on mobile and `clear` button on the same line with the last one. We'll whether decline this design or go and implement again some stuff in the table-filter component implementing this particular case.

Reply:
I really don't want go this way. I have never seen a library of reusable components where you can pass classes to components to override their behaviour or layout or styles. It's responsibility to ec-table-filter to lay down components inside, having a class to override that means we have to provide these classes everywhere we use ec-table-filter which means they are going to break every time we refactor something inside ec-table-filter. It's going to be maintenance nightmare. Follow the YAGNI and don't try to optimise/support for cases we don't have.